feat: makes quadlet content optional

axel7083 opened the pull request:
## Description

When a Quadlet is in an error state (invalid YAML, invalid property, etc.) it does not have an associated systemd service.

The `Quadlet#content` property corresponds to the systemd generate content, but when the quadlet is in an error state this is undefined.

To better reflect the reality, making the field optional, and nicely handle it.

## Screenshots

**Quadlet in Error state**

![image](https://github.com/user-attachments/assets/3db76a15-f4d3-474c-9146-7d26babe41e1)

**Valid Quadlet**

![image](https://github.com/user-attachments/assets/87e61134-209f-4c51-b8a8-c6e9137c94f6)


## Related issues

Fixes https://github.com/podman-desktop/extension-podman-quadlet/issues/343

## Testing

- [x] unit tests has been provided

jeffmaury reviewed (commented):
Looks to me there are changes in this PR that are not related to the description/concern

axel7083 commented:
> Looks to me there are changes in this PR that are not related to the description/concern

@jeffmaury yes some tests are covering more than the PR changes, but there is no feature out of scope.

## Details

Making the content optional force us to handle the undefined case in some places

- In `packages/backend/src/services/quadlet-service.ts` I will throw an error is the content is undefined, and we try to parse it
- In `packages/backend/src/utils/parsers/quadlet-dryrun-parser.ts` I return undefined instead of an empty string 

On the frontend, I updated the tabs, to not show the content if it is undefined, and added a test to ensure this change works

gastoner reviewed (approved):
LGTM on W
